extensions/avro-extensions/src/main/java/org/apache/parquet/avro/DruidAvroReadSupport.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.avro;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import io.druid.indexer.HadoopDruidIndexerConfig;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.CountAggregatorFactory;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.util.ReflectionUtils;
+import org.apache.parquet.hadoop.api.InitContext;
+import org.apache.parquet.io.api.RecordMaterializer;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class DruidAvroReadSupport extends AvroReadSupport<GenericRecord>
+{
+  private MessageType getPartialReadSchema(InitContext context)
+  {
+    MessageType fullSchema = context.getFileSchema();
+
+    String name = fullSchema.getName();
+
+    HadoopDruidIndexerConfig config = HadoopDruidIndexerConfig.fromConfiguration(context.getConfiguration());
+    String tsField = config.getParser().getParseSpec().getTimestampSpec().getTimestampColumn();
+    List<String> dimensions = config.getParser().getParseSpec().getDimensionsSpec().getDimensions();
+    Set<String> metricsFields = Sets.newHashSet();
+    for (AggregatorFactory agg : config.getSchema().getDataSchema().getAggregators()) {
+      if(agg instanceof CountAggregatorFactory == false){
+        metricsFields.addAll(agg.requiredFields());
+      }
+    }
+
+    List<Type> partialFields = Lists.newArrayList();
+
+    for (Type type : fullSchema.getFields()) {
+      if (tsField.equals(type.getName())
+          || metricsFields.contains(type.getName())
+          || dimensions.size() > 0 && dimensions.contains(type.getName())
+          || dimensions.size() == 0) {
+        partialFields.add(type);
+      }
+    }
```

Comment:
do you know if, in ParquetInputFormat,  conversion from "ORC record" to "Generic record" is lazy or eager? If it is lazy then this shouldn't matter, actual conversion would only happen for the fields that have been explicitly asked for.